File: board.rb
# encoding: utf-8

require_relative "./tile"

class Board

  attr_reader :board

  def initialize
    @board = Array.new(9) { |row| Array.new(9){ |col| Tile.new(self, row, col)}}
    seed_bombs
    @blew_up = false
  end 

  def seed_bombs
    10.times do
      tile = @board.sample.sample

      while tile.bombed?
        tile = @board.sample.sample
      end

      tile.bomb
    end 
  end 

  def display
    @board.each do |row|
      display_row = []

      row.each do |tile|
        display_row << tile.display
      end 

      p display_row.join("   ")
      puts
    end 
  end 

  def display_actual
    @board.each do |row|
      display_row = []

      row.each do |tile|
        display_row << tile.true_display
      end #each

      p display_row.join("   ")
      puts
    end #each
  end #display_actual

  def reveal(row, col)
    @board[row][col].reveal
    @blew_up = @board[row][col].bombed?
  end #reveal

  def flag(row, col)
    @board[row][col].flag
  end #flag

  def lost?
    return @blew_up
  end #lost?

  def won?
    all_tiles = []

    @board.each do |row|
      row.each do |tile|
        all_tiles << tile
      end
    end

    unrevealed_tiles = all_tiles.select { |tile| !tile.revealed?}
    unrevealed_tiles.count == 10
  end 
end 

File: board.js
(function (root) {
	var Minesweeper = root.Minesweeper = (root.Minesweeper || {});

	var Board = Minesweeper.Board = function(dimension) {
		this.dimension = dimension;
		this.board = generateBoard(dimension);
		this.generateBoardDisplay();
		this.blew_up = false;
		this.seedBombs;
	};

	var Board.prototype.generateBoard = function(dimension) {
		var rows = new Array (dimension);
		for (i = 0; i < rows.length; i++) {
			rows[i] = new Array (dimension)
			for (j = 0; j < rows.length; j++) {
				rows[i][j] = new Minesweeper.Tile(this, i, j);
			}
		}
		return rows;
	};

	var Board.prototype.generateBoardDisplay = function() {
		var container = document.getElementById('boardContainer');
		for (i = 0; i < this.length; i++) {
			//insert a div for row this[i].html('<div></div>')
			for (j = 0; j < this.length; j++) {
				//insert a div for each tile this[i][j].display
			}
		}
	};

	var Board.prototype.seedBombs = function() {
		for (i = 0; i < this.dimension; i++) {
			var randRow = Math.floor(Math.random()*10)
			var randCol = Math.floor(Math.random()*10)
			var tile = this[randRow][randCol]

			while (tile.bombed?) {
				var randRow = Math.floor(Math.random()*10)
				var randCol = Math.floor(Math.random()*10)
				var tile = this[randRow][randCol]
			}

			tile.bomb();
		}	
	};

	var Board.prototype.reveal = function(row, col) {
		this[row][col].reveal;
    	this.blew_up = this[row][col].bombed?
	};

	var Board.prototype.flag = function() {
		this[row][col].flag;
	};

	var Board.prototype.won? = function() {
		var all_tiles = [];

		for (i = 0; i < this.length; i++) {
			for (j = 0; j < this.length; j++) {
				all_tiles.push(this[i][j]);
			}
		}

		var unrevealed_tiles = all_tiles.filter(function(tile) {return !tile.revealed?})
		unrevealed_tiles.count === 10
	};

	var Board.prototype.lost? = function() {
		return this.blew_up
	}

})(this);